lib/authentication/auth_okta.js

```diff
@@ -164,13 +164,20 @@ function AuthOkta(connectionConfig, httpClient) {
   *
   * @returns {null}
   */
-  function validateURLs(authenticator, ssoUrl, tokenUrl) {
-    authenticator = authenticator.toLowerCase();
-    if (!(authenticator.startsWith(ssoUrl.substring(0, authenticator.length)) &&
-      authenticator.startsWith(tokenUrl.substring(0, authenticator.length)))) {
-      throw new Error('The prefix of the SSO/token URL and the specified authenticator do not match.');
+  this.validateURLs = function (authenticator, ssoUrl, tokenUrl) {
+    try {
+      const aUrl = new URL(authenticator);
+      const sUrl = new URL(ssoUrl);
+      const tUrl = new URL(tokenUrl);
+      if (!(compareUrlsByProtocolAndHost(aUrl, sUrl) && compareUrlsByProtocolAndHost(aUrl, tUrl))) {
+        throw new Error('The prefix of the SSO/token URL and the specified authenticator do not match.');
+      }
+    } catch (err) {
```

Comment:
by catching here and throwing a different exception we are loosing the exception from error above? should we pass cause with error?

Reply:
a very fair point, my bad. 
problem is that `error.cause` is convenient but unfortunately [not available until node16](https://nodejs.org/docs/latest-v16.x/api/errors.html#errorcause) and while we need to maintain compatibility with node14, had to do something else.

Reply:
Looks and working correct.